src/SBInclinedExponential.cpp

```diff
@@ -0,0 +1,367 @@
+        // count for the number of required digits, which is nominally given by digits10.
+        // cf. http://stackoverflow.com/questions/4738768/printing-double-without-losing-precision
+        // Unfortunately, there doesn't seem to be an easy equivalent of python's %r for
+        // printing the repr of a double that always works and doesn't print more digits than
+        // it needs.  This is the reason why we reimplement the __repr__ methods in python
+        // for all the SB classes except SBProfile.  Only the last one can't be done properly
+        // in python, so it will use the C++ virtual function to get the right thing for
+        // any subclass.  But possibly with ugly extra digits.
+        oss.precision(std::numeric_limits<double>::digits10 + 4);
+        oss << "galsim._galsim.SBInclinedExponential("<<getI()<<", "<<getScaleRadius()<<", "<<getScaleHeight();
+        oss <<", "<<getFlux()<<", False";
+        oss << ", galsim.GSParams("<<*gsparams<<"))";
+        return oss.str();
+    }
+
+    LRUCache< boost::tuple<double, GSParamsPtr >, InclinedExponentialInfo >
+        SBInclinedExponential::SBInclinedExponentialImpl::cache(sbp::max_inclined_exponential_cache);
+
+    SBInclinedExponential::SBInclinedExponentialImpl::SBInclinedExponentialImpl(double i, double scale_radius, double scale_height, double flux,
+            const GSParamsPtr& gsparams) :
+        SBProfileImpl(gsparams),
+        _r0(scale_radius),
+        _h0(scale_height),
+        _flux(flux),
+        _inv_r0(1./_r0),
+        _h_tani_over_r(scale_height*std::tan(i)/scale_radius),
+        _r0_cosi(_r0*std::cos(i)),
+        _inv_r0_cosi(1./(scale_radius*std::cos(i))),
+        // Start with InclinedExponentialInfo regardless of value of trunc
+        _info(cache.get(boost::make_tuple(_h_tani_over_r, this->gsparams.duplicate())))
+    {
+        dbg<<"Start SBInclinedExponential constructor:\n";
+        dbg<<"i = "<<_i<<std::endl;
+        dbg<<"scale radius = "<<_r0<<std::endl;
+        dbg<<"scale height = "<<_h0<<std::endl;
+        dbg<<"flux = "<<_flux<<std::endl;
+
+        // Ensure i is positive and in the range 0 <= i <= M_PI/2
+
+        // Work with positive values first
+        if(i<0) i = -i;
+
+        // Get into the proper range
+        if(i>M_PI/2.)
+        {
+            i -= int(i/(2*M_PI))*2*M_PI;
+            if(i<0) i = -i; // Since we might have gone below zero again
+        }
+
+        // Check if i is within allowed limits, and institute special handling if it isn't
+        if(i>sbp::maximum_i)
+        {
+            // Perfectly edge-on isn't analytic, so we truncate at the maximum value
+            _i = sbp::maximum_i;
+        }
+        else
+        {
+            // Normal setup here
+            _i = i;
+            _h_tani_over_r = scale_height*std::tan(i)/scale_radius;
+            _r0_cosi = _r0*std::cos(i);
+            _inv_r0_cosi = 1./(scale_radius*std::cos(i));
+        }
```

Comment:
A few points about this bit.

1. the logic here is a bit wrong.  You recompute `_h_tani_over_r`, `_r0_cosi`, and `_inv_r0_cosi` when `i` is within the allowed limits, so the calculation you did in the initializer list is still good.  But in the case where you changed `_i`, you don't recalculate, even though they are incorrect.

2. I'd recommend changing the limit to be on `cos(i)` rather than `i`.  The problem with `i` near pi/2 is that cos(i) == 0, so `inv_r0_cosi` is undefined.  Probably better to just set a minimum on `cosi` instead.

3. I don't think you need all the adjustments of the range then.  Just take the absolute value of cos(i) and tan(i).

4. Once `i` is switch to being an `Angle`, note that you can do the trig calls with `i.cos()` in lieu of `std::cos(i.rad())`.  It looks like we never wrote free functions `cos(i)`, etc. for `Angle` arguments.  That would be even nicer looking and of course easy to do, so if you feel like adding those, please go ahead.